pandas/core/arrays/datetimelike.py

```diff
@@ -197,6 +201,27 @@ def astype(self, dtype, copy=True):
             return self._box_values(self.asi8)
         return super(DatetimeLikeArrayMixin, self).astype(dtype, copy)
 
+    # ------------------------------------------------------------------
+    # Formatting
+
+    def __repr__(self):
+        if len(self) <= 6:
+            # TODO: 6 is an arbitrary cutoff; make configurable?
+            items = [str(s) for s in self]
+        else:
```

Comment:
pls don’t repeat this code

we have extensive support already for all of this (and it is heavily tested)
in Index (it actually is not restricted there) and relies on
pandas.io.formats

Reply:
https://github.com/pandas-dev/pandas/blob/master/pandas/io/formats/printing.py#L275

Reply:
what do you have in mind?  Aside from possibly-truncating, this is just the existing PeriodArray implementation

Reply:
and that is repeating unecessary code pls make the change

Reply:
https://github.com/pandas-dev/pandas/blob/master/pandas/core/indexes/base.py#L1049

Reply:
Reusing with indexes is going to be difficult. That's templated and dispatches to a bunch of methods which don't exist on EAs. Last time I looked at consolidating them it didn't seem worth it.

Reply:
i am not suggesting reusing if but reusing the code it calls which is completely generic

pls do not reinvent the already solved wheel here

i don’t recall this from PeriodArray
but it needs to change either here or as a precursor

Reply:
Almost there:

```python
In [39]: print(pd.io.formats.printing.format_object_summary(arr, str, is_justify=False, name=''))
[1970-01-01, 1970-01-02, 1970-01-03, 1970-01-04, 1970-01-05, 1970-01-06,
  1970-01-07, 1970-01-08, 1970-01-09, 1970-01-10],
```

vs.

```python
In [38]: arr
Out[38]:
<PeriodArray>
['1970-01-01', '1970-01-02', '1970-01-03', '1970-01-04', '1970-01-05', '1970-01-06', '1970-01-07', '1970-01-08', '1970-01-09', '1970-01-10']
Length: 10, dtype: period[D]
```

longer:

```python
In [44]: arr = pd.core.arrays.PeriodArray(np.arange(10000), freq='D')

In [45]: print(pd.io.formats.printing.format_object_summary(arr, str, is_justify=False, name=''))
[1970-01-01, 1970-01-02, 1970-01-03, 1970-01-04, 1970-01-05, 1970-01-06,
  1970-01-07, 1970-01-08, 1970-01-09, 1970-01-10,
  ...
  1997-05-09, 1997-05-10, 1997-05-11, 1997-05-12, 1997-05-13, 1997-05-14,
  1997-05-15, 1997-05-16, 1997-05-17, 1997-05-18],

```

Is there an off by one error in the printing?

Reply:
Opened #23590 to follow up on this

Reply:
no I would do this as a pre-cursor rather than a followup (or here).

Reply:
On the call there was discussion of a 1-week goal for getting all this done.  I can't keep up the amount of time I've been putting in on this.  A precursor to this PR is a non-starter.  So the options right now are this implementation or removing this entirely, leaving us with a strictly-worse implementation for PeriodArray and none of the tests.

Reply:
I can followup on it later today (now).

Reply:
well u spent a lot of time adding code which now has to be fixed

i really really don’t like reinventing the wheel

ok to merge but this must be followed up and fixed 

maybe smaller prs are better

Reply:
Not a big deal. I've been meaning to do one for the base EA class anyway, and will have it done shortly.